### PR TITLE
Delay WebExtension startup

### PR DESCRIPTION
**Pull request by 6a68**
See #2575 for background

### CODE REVIEW

**Comment by 6a68** [comment]
Here's a Try job to run the browser chrome tests with this patch. We'll see if anything breaks that was green. https://treeherder.mozilla.org/#/jobs?repo=try&revision=351a6a0120b4ea0b3ef2d621af676a9d6e8a7452

File: addon/bootstrap.js [hunk 1]
```diff
@@ -30,15 +39,31 @@ const prefObserver = {
...
+  },
+
+  observe: function() {
+    appStartupDone();
+    this.unregister();
+  }
+}
+
 function startup(data, reason) { // eslint-disable-line no-unused-vars
+  appStartupObserver.register();
```
Comment (ianb) [inline addon/bootstrap.js]: Are we confident that there's no way this code could be called after sessionstore-windows-restored is fired?  Like... if the add-on is reloaded?

**Comment by 6a68** [comment]
The first try job was on central, here's one on inbound:

https://treeherder.mozilla.org/#/jobs?repo=try&revision=7f60640f1ce9617d9ec00167a7eda75e9d647ffc

File: addon/bootstrap.js (hunk 1 above)
Comment (6a68) [inline addon/bootstrap.js]: Great question. I think I should add a check that the `reason` is startup, and not upgrade

Comment (ianb) [inline addon/bootstrap.js]: Even that seems a little fragile.  Is there a flag that changes when that event fires, so we can tell if the event won't fire?  Kind of like a readyState?

**Comment by 6a68** [comment]
Thanks for the review, @ianb. Now resolving the promise unless the startup `reason` is APP_STARTUP. Merging

**Pull request merged**